### ISSUE TITLE
Duplicate Validation Rules

### ISSUE THREAD

**Issue by astormcupdx** [intrip/laravel-authentication-acl#46]
I'm not sure if this is something you need/want to address, but it's a buggy thing I ran into using the 1.2 branch of the project.

I was writing some import scripts, and had some code where the `Jacopo\Authentication\Validators\UserValidator` class was instantiated over and over again in a loop.  This uncovered some weird behavior with the `__constructor` method.

```
#File: vendor\jacopo\laravel-authentication-acl\src\Jacopo\Authentication\Validators\UserValidator.php    public function __construct()
{
    Event::listen('validating', function($input)
    {
        // check if the input comes form the correct form
        if(!isset($input['form_name']) || $input['form_name']!='user')
            return true;

        if(empty($input["id"]))
        {
            static::$rules["password"][] = "required";
            static::$rules["email"][] = "unique:users,email";
        }
        else
        {
            static::$rules["email"][] = "unique:users,email,{$input['id']}";
        }
    });
}
```

Whenever the `UserValidator` object is instantiated, it adds rules to the static `$rules` array.  When you call this in a loop, it means a set of duplicate rules is added over and over again.  This means the Laravel validator needs to validate an increasingly long list of rules and you get an O(n) performance problem (i.e. -- stuff gets slower and slower the more you run through the loop).

I was able to fix the problem by ensuring my code only instantiated the a `UserValidator` once and then reused the object. However, it did take an hour or so of debugging to figure out everything that was going on.  

Something like the following code at the end of the constructor

```
static::$rules["email"] = array_unique(static::$rules["email"]);
static::$rules["password"] = array_unique(static::$rules["password"]);
```

would ensure that duplicate validate rules were removed, and mean future users won't run into the same problem.

**Comment by intrip** [comment]
You made a nice point i've added your fix to the 1.2 branch also i'll remove the bad static handling of the rules and make them instantiated classes in the next future.

**Comment by intrip** [comment]
I'll push the fix when i'm done with some other stuff because test are failing now.
